nwbn_conversion_tools/ephys/intan/intan.py

```diff
@@ -0,0 +1,157 @@
+        meta_electrode_groups = metadata_ecephys['ElectrodeGroup']
+        for meta in meta_electrode_groups:
+            self.nwbfile.create_electrode_group(
+                name=meta['name'],
+                description=meta['description'],
+                location=meta['location'],
+                device=device
+            )
+
+        # Gets electrodes info from first rhd file
+        file_data = load_intan.read_data(filename=all_files_rhd[0])
+        electrodes_info = file_data['amplifier_channels']
+        n_electrodes = len(electrodes_info)
+
+        # Electrodes
+        if electrodes_file is not None:  # if an electrodes info file was provided
+            df_electrodes = pd.read_csv(electrodes_file, index_col='Channel Number')
+            for idx, elec in enumerate(electrodes_info):
+                elec_name = elec['native_channel_name']
+                elec_group = df_electrodes.loc[elec_name]['electrode_group']
+                elec_imp = df_electrodes.loc[elec_name]['Impedance Magnitude at 1000 Hz (ohms)']
+                self.nwbfile.add_electrode(
+                    id=idx,
+                    x=np.nan, y=np.nan, z=np.nan,
+                    imp=float(elec_imp),
+                    location='location',
+                    filtering='none',
+                    group=self.nwbfile.electrode_groups[elec_group]
+                )
+        else:  # if no electrodes file info was provided
+            first_el_grp = list(self.nwbfile.electrode_groups.keys())[0]
+            electrode_group = self.nwbfile.electrode_groups[first_el_grp]
+            for idx in range(n_electrodes):
+                self.nwbfile.add_electrode(
+                    id=idx,
+                    x=np.nan, y=np.nan, z=np.nan,
+                    imp=np.nan,
+                    location='location',
+                    filtering='none',
+                    group=electrode_group
+                )
+
+        electrode_table_region = self.nwbfile.create_electrode_table_region(
+            region=list(np.arange(n_electrodes)),
+            description='no description'
+        )
+
+        # Create iterator
+        data_iter = DataChunkIterator(
+            data=data_gen(source_dir=dir_ecephys_rhd),
+            iter_axis=0,
+            buffer_size=10000,
+            maxshape=(None, n_electrodes)
+        )
+
+        # Electrical Series
+        # Gets electricalseries conversion factor
+        es_conversion_factor = file_data['amplifier_data_conversion_factor']
+        ephys_ts = ElectricalSeries(
+            name=metadata_ecephys['ElectricalSeries'][0]['name'],
+            description=metadata_ecephys['ElectricalSeries'][0]['description'],
+            data=data_iter,
+            electrodes=electrode_table_region,
+            rate=sampling_rate,
+            starting_time=0.0,
+            conversion=es_conversion_factor
+        )
+        self.nwbfile.add_acquisition(ephys_ts)
```

Comment:
It seems like most of this is not specific to intan but general extracellular electrophysiology code. Remember, we want to minimize repeating code. If you think some of these lines might be used by another acquisition system, is there a way to make a general ecephys conversion class that holds some of this? It seems like this is more general and could be in `NWBEcephysConverter` and then have `NWBConverter`>`NWBEcephysConverter`>`Intan2NWB`.